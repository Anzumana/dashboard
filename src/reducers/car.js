const initialState = {
};
export default (state = initialState, action) => {
  switch (action.type) {
    case '[CAR]:SET':
      return action.payload;

    default:
      return state;
  }
};
