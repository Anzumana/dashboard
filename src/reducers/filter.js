const initialState={
	since:'',
	until:''
}
export default (state = initialState, action) =>{
	switch(action.type){

		case 'SET_FILTEr':
			return action.payload;

		default:
			return state;
			
	}
}
