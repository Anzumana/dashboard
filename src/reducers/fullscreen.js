const initalState = false;

export default function(state = initalState , action){
	switch(action.type){
		case 'Fullscreen:on':
			return true;
		case 'Fullscreen:off':
			return false;
		case 'Fullscreen:toggle':
			return !state;
		default:
			return state;
	}
};
