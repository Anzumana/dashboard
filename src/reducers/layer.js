const initialState = null;
export default function (state = initialState, action) {
  switch (action.type) {
    case '[LAYER]:typically_affected_subgraph':
      return 'typically_affected_subgraph';
    case '[LAYER]:affected_subgraph':
      return 'affected_subgraph';
    case '[LAYER]:none':
      return 'none';
    default:
      return state;
  }
}
