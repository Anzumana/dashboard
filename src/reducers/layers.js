const initalState= {
	activeOptions:{
	},
	mapOptions:{
		'streets': { id: 'streets' , path: 'Take out the garbage' },
		'outdoors': { id: 'outdoors' , path: 'Watch my favorite show' },
		'dark': { id: 'dark' , path: 'Take out the garbage' },
		'light': { id: 'light' , path: 'Charge my phone' },
		'satelliteStreets': { id: 'satelliteStreets' , path: 'Cook dinner' },
		'satelliteStreets': { id: 'satelliteStreets', path: 'test path' },
		'navigationPreviewDay': { id: 'navigationPreviewDay', path: 'test path' },
		'navigationPreviewNight': { id: 'navigationPreviewNight', path: 'test path' },
		'navigationGuidanceDay': { id: 'navigationGuidanceDay', path: 'test path' },
		'navigationGuidanceNight': { id: 'navigationGuidanceNight', path: 'test path' },
	},
	dataOptions:{
		 'affected': { id: 'affected' , path: 'Take out the garbage' },
		 'typicall': { id: 'typicall' , path: 'Take out the garbage' },
	},
	columns:{
		'activeOptions': {
			id: 'activeOptions',
			title: 'Map Options',
			taskIds: [] 
		},
		'mapOptions': {
			id: 'mapOptions',
			title: 'Data Options',
			taskIds: [
				"streets",
				"outdoors",
				"light",
				"dark",
				"satellite",
				"satelliteStreets",
				"navigationPreviewDay",
				"navigationPreviewNight",
				"navigationGuidanceDay",
				"navigationGuidanceNight",
			]
		},
		'dataOptions': {
			id: 'dataOptions',
			title: 'Data Options',
			taskIds: ['typicallyAffectedSubgraph','affectedSubgraph'] 
		}
	},
	columnOrder: ['activeOptions','mapOptions','dataOptions']
};

export default function(state = initalState , action){
	switch(action.type){
		case 'REORDER':
			return action.payload;
		default:
			return state;
	}
};
