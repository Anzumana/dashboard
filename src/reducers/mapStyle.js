const initalState= "mapbox://styles/mapbox/streets-v10";

export default function(state = initalState , action){
			let value = '';
	switch(action.type){
			case 'typicallyAffectedSubgraph' :
				break;
			case 'affectedSubgraph':
				break;
			case	"[mapStyle] street":
				value = "mapbox/streets-v10";
				return 	"mapbox://styles/" + value
			case "[mapStyle] outdoors":
				value = "mapbox/outdoors-v10";
				return "mapbox://styles/" + value
			case "[mapStyle] light":
				value = "mapbox/light-v9";
				return "mapbox://styles/" + value
			case "[mapStyle] dark":
				value = "mapbox/dark-v9";
				return "mapbox://styles/" + value
			case "[mapStyle] satellite":
				value = "mapbox/satellite-v9";
				return "mapbox://styles/" + value
			case "[mapStyle] satelliteStreets":
				value = "mapbox/satellite-streets-v10";
				return "mapbox://styles/" + value
			case "[mapStyle] navigationPreviewDay":
				value = "mapbox/navigation-preview-day-v4";
				return "mapbox://styles/" + value;
			case "[mapStyle] navigationPreviewNight":
				value = "mapbox/navigation-preview-night-v4";
				return "mapbox://styles/" + value;
			case "[mapStyle] navigationGuidanceDay":
				value = "mapbox/navigation-guidance-day-v4";
				return 'mapbox://styles/' + value
			case "[mapStyle] navigationGuidanceNight":
				value = "mapbox/navigation-guidance-night-v4";
				return "mapbox://styles/" + value;
		default:
			return state;
	}
};
