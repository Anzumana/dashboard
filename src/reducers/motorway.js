		const initalState = []
export default function(state = initalState, action){
	switch(action.type){
		case '[MOTORWAY]:SET':
			return action.payload;
		default:
			 return state;
	}
}
