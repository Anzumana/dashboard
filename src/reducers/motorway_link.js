const initalState = [];
export default function (state = initalState, action) {
  switch (action.type) {
    case '[MOTORWAY_LINK]:SET':
      return action.payload;
    default:
      return state;
  }
}
