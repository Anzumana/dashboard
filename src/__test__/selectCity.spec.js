import reducers from '../reducers/selectCity';

test('selectCity reducer init', () => {
  const result = reducers(undefined, {type:'ANYTHING'});
	expect(result).toBeDefined();
});

test('set viewport', () => {
	const startState = {
			    city: '----',
				    cities: [
							      {
											        text: 'Hannover',
											        value: 'hannover',
											        location: '9.548431,52.470767-9.973886,52.296619',
											        viewport: {
																          latitude: 52.376712,
																          longitude: 9.75426,
																          zoom: 9,
																          pitch: 58.07993019329368,
																          bearing: 10.55792307692307,
																          transitionDuration: 3000
																        }
											      },
							      {
											        text: 'Braunschweig',
											        value: 'braunschweig',
											        location: '10.355531,52.361515-10.727006,52.152200',
											        viewport: {
																          latitude: 52.275471,
																          longitude: 10.524445,
																          zoom: 9,
																          pitch: 58.07,
																          bearing: 10.5579,
																          transitionDuration: 3000
																        }
											      },
							      {
											        text: 'Wolfsburg',
											        value: 'wolfsburg',
											        location: '10.611414,52.489153-10.967783,52.291364',
											        viewport: {
																          latitude: 52.41383,
																          longitude: 10.773462,
																          zoom: 9,
																          pitch: 58.59013474066599,
																          bearing: -49.63659663812949,
																          transitionDuration: 3000
																        }
											      }
							    ]
			  };
	const expectedState= {
		     city: 'hannover',
			     cities: [
						       {
										         text: 'Hannover',
										         value: 'hannover',
										         location: '9.548431,52.470767-9.973886,52.296619',
										         viewport: {
															           latitude: 52.376712,
															           longitude: 9.75426,
															           zoom: 9,
															           pitch: 58.07993019329368,
															           bearing: 10.55792307692307,
															           transitionDuration: 3000
															         }
										       },
						       {
										         text: 'Braunschweig',
										         value: 'braunschweig',
										         location: '10.355531,52.361515-10.727006,52.152200',
										         viewport: {
															           latitude: 52.275471,
															           longitude: 10.524445,
															           zoom: 9,
															           pitch: 58.07,
															           bearing: 10.5579,
															           transitionDuration: 3000
															         }
										       },
						       {
										         text: 'Wolfsburg',
										         value: 'wolfsburg',
										         location: '10.611414,52.489153-10.967783,52.291364',
										         viewport: {
															           latitude: 52.41383,
															           longitude: 10.773462,
															           zoom: 9,
															           pitch: 58.59013474066599,
															           bearing: -49.63659663812949,
															           transitionDuration: 3000
															         }
										       }
						     ]
		   };
	const result = reducers(startState, 
			{
				type:'SELECT_CITY', 
				payload:'hannover'
				
			});
	expect(result).toEqual(expectedState);
});
