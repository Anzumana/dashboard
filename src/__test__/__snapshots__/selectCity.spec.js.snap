// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`set viewport 1`] = `
Object {
  "cities": Array [
    Object {
      "location": "9.548431,52.470767-9.973886,52.296619",
      "text": "Hannover",
      "value": "hannover",
      "viewport": Object {
        "bearing": 10.55792307692307,
        "latitude": 52.376712,
        "longitude": 9.75426,
        "pitch": 58.07993019329368,
        "transitionDuration": 3000,
        "zoom": 9,
      },
    },
    Object {
      "location": "10.355531,52.361515-10.727006,52.152200",
      "text": "Braunschweig",
      "value": "braunschweig",
      "viewport": Object {
        "bearing": 10.5579,
        "latitude": 52.275471,
        "longitude": 10.524445,
        "pitch": 58.07,
        "transitionDuration": 3000,
        "zoom": 9,
      },
    },
    Object {
      "location": "10.611414,52.489153-10.967783,52.291364",
      "text": "Wolfsburg",
      "value": "wolfsburg",
      "viewport": Object {
        "bearing": -49.63659663812949,
        "latitude": 52.41383,
        "longitude": 10.773462,
        "pitch": 58.59013474066599,
        "transitionDuration": 3000,
        "zoom": 9,
      },
    },
  ],
  "city": "hannover",
}
`;
